### PR TITLE
Update jaxb implementation

### LINKED ISSUE #142: Exception when trying to use completion function from a beta runtime artifact (JAXB-related)
Recreate -
1. Install Liberty Tools Eclipse  (Early Release)
2. Configure LMP to use beta runtime artifact 
```xml
                <configuration>
                    <runtimeArtifact>
                    <groupId>io.openliberty.beta</groupId>
                    <artifactId>openliberty-runtime</artifactId>
                    <version>LATEST</version>
                    <type>zip</type>
                    </runtimeArtifact>
                </configuration>
```
3. Do a `mvn liberty:create` (otherwise the runtime will be irrelevant and we'll go down the default feature path)
4. Open a server.xml, position the cursor with the contents of a `<feature>xyz-1.0</feature>` element and type `Ctrl+Space`.

In the debugger I see NullPointerException thrown here:
```
Injector.inject(String, byte[]) line: 294	
Injector.inject(ClassLoader, String, byte[]) line: 66	
AccessorInjector.prepare(Class, String, String, String...) line: 57	
OptimizedAccessorFactory.get(Field) line: 157	
Accessor$FieldReflection<BeanT,ValueT>.optimize(JAXBContextImpl) line: 255	
SingleElementNodeProperty<BeanT,ValueT>.<init>(JAXBContextImpl, RuntimeElementPropertyInfo) line: 64	
NativeConstructorAccessorImpl.newInstance0(Constructor<?>, Object[]) line: not available [native method]	
NativeConstructorAccessorImpl.newInstance(Object[]) line: 77	
DelegatingConstructorAccessorImpl.newInstance(Object[]) line: 45	
Constructor<T>.newInstanceWithCaller(Object[], boolean, Class<?>) line: 499	
Constructor<T>.newInstance(Object...) line: 480	
PropertyFactory.create(JAXBContextImpl, RuntimePropertyInfo) line: 99	
ClassBeanInfoImpl<BeanT>.<init>(JAXBContextImpl, RuntimeClassInfo) line: 150	
JAXBContextImpl.getOrCreate(RuntimeClassInfo) line: 493	
JAXBContextImpl.getOrCreate(RuntimeTypeInfo) line: 512	
ArrayElementNodeProperty<BeanT,ListT,ItemT>(ArrayElementProperty<BeanT,ListT,ItemT>).<init>(JAXBContextImpl, RuntimeElementPropertyInfo) line: 82	
ArrayElementNodeProperty<BeanT,ListT,ItemT>.<init>(JAXBContextImpl, RuntimeElementPropertyInfo) line: 32	
NativeConstructorAccessorImpl.newInstance0(Constructor<?>, Object[]) line: not available [native method]	
NativeConstructorAccessorImpl.newInstance(Object[]) line: 77	
DelegatingConstructorAccessorImpl.newInstance(Object[]) line: 45	
Constructor<T>.newInstanceWithCaller(Object[], boolean, Class<?>) line: 499	
Constructor<T>.newInstance(Object...) line: 480	
PropertyFactory.create(JAXBContextImpl, RuntimePropertyInfo) line: 99	
ClassBeanInfoImpl<BeanT>.<init>(JAXBContextImpl, RuntimeClassInfo) line: 150	
JAXBContextImpl.getOrCreate(RuntimeClassInfo) line: 493	
JAXBContextImpl.<init>(JAXBContextImpl$JAXBContextBuilder) line: 310	
JAXBContextImpl.<init>(JAXBContextImpl$JAXBContextBuilder, JAXBContextImpl$1) line: 109	
JAXBContextImpl$JAXBContextBuilder.build() line: 1142	
ContextFactory.createContext(Class[], Map<String,Object>) line: 141	
NativeMethodAccessorImpl.invoke0(Method, Object, Object[]) line: not available [native method]	
NativeMethodAccessorImpl.invoke(Object, Object[]) line: 77	
DelegatingMethodAccessorImpl.invoke(Object, Object[]) line: 43	
Method.invoke(Object, Object...) line: 568	
ContextFinder.newInstance(Class[], Map, String) line: 211	
ContextFinder.find(Class[], Map) line: 372	
JAXBContext.newInstance(Class[], Map<String,?>) line: 574	
JAXBContext.newInstance(Class...) line: 522	
FeatureService.getInstalledFeaturesList(String, String, String) line: 251	
FeatureService.getFeatures(String, String, int, String) line: 183	
FeatureService.getFeature(String, String, String, int, String) line: 194	
LibertyHoverParticipant.getHoverFeatureDescription(String, DOMDocument) line: 70	
LibertyHoverParticipant.onText(IHoverRequest) line: 57	
XMLHover.getTextHover(HoverRequest, Range) line: 207	
XMLHover.doHover(DOMDocument, Position, SharedSettings, CancelChecker) line: 99	
XMLLanguageService.doHover(DOMDocument, Position, SharedSettings, CancelChecker) line: 167	
XMLTextDocumentService.lambda$hover$5(HoverParams, DOMDocument, CancelChecker) line: 260	
0x000000080047e230.apply(Object, Object) line: not available	
ModelTextDocuments<T>.lambda$computeModelAsync$0(TextDocumentIdentifier, BiFunction, CancelChecker) line: 116	
0x000000080016c590.apply(Object) line: not available	
CompletableFuture$UniApply<T,V>.tryFire(int) line: 646	
CompletableFuture$UniApply<T,V>(CompletableFuture$Completion).exec() line: 483	
CompletableFuture$UniApply<T,V>(ForkJoinTask<V>).doExec() line: 373	
ForkJoinPool$WorkQueue.topLevelExec(ForkJoinTask<?>, WorkQueue) line: 1182	
ForkJoinPool.scan(ForkJoinPool$WorkQueue, int, int) line: 1655	
ForkJoinPool.runWorker(ForkJoinPool$WorkQueue) line: 1622	
ForkJoinWorkerThread.run() line: 165	
```

which gets caught in stack 
> java.lang.NullPointerException: Cannot invoke "java.lang.reflect.Method.invoke(Object, Object[])" because "com.sun.xml.bind.v2.runtime.reflect.opt.Injector.defineClass" is null

caught in stack:
```
XMLHover.getTextHover(HoverRequest, Range) line: 212	
XMLHover.doHover(DOMDocument, Position, SharedSettings, CancelChecker) line: 99	
XMLLanguageService.doHover(DOMDocument, Position, SharedSettings, CancelChecker) line: 167	
XMLTextDocumentService.lambda$hover$5(HoverParams, DOMDocument, CancelChecker) line: 260	
0x000000080047e230.apply(Object, Object) line: not available	
ModelTextDocuments<T>.lambda$computeModelAsync$0(TextDocumentIdentifier, BiFunction, CancelChecker) line: 116	
0x000000080016c590.apply(Object) line: not available	
CompletableFuture$UniApply<T,V>.tryFire(int) line: 646	
CompletableFuture$UniApply<T,V>(CompletableFuture$Completion).exec() line: 483	
CompletableFuture$UniApply<T,V>(ForkJoinTask<V>).doExec() line: 373	
ForkJoinPool$WorkQueue.topLevelExec(ForkJoinTask<?>, WorkQueue) line: 1182	
ForkJoinPool.scan(ForkJoinPool$WorkQueue, int, int) line: 1655	
ForkJoinPool.runWorker(ForkJoinPool$WorkQueue) line: 1622	
ForkJoinWorkerThread.run() line: 165	
```

Note: we are running the LCLS with Java 17 in Liberty Tools Eclipse.

### PR DESCRIPTION
**Pull request by evie-lau**
Fixes #142 

The initial load seems to take a while when trying to run completion on a server.xml `<feature>` tag, but it works now

### CODE REVIEW

**Review by cherylking — CHANGES_REQUESTED** [review]
I'm glad the fix is straight forward! Two things though:

1) Update copyrights to include 2023.
2) Can a test be updated for this to prove it works?

**Comment by scottkurz** [comment]
@evie-lau just curious...how was this tested so far?

**Comment by evie-lau** [comment]
> @evie-lau just curious...how was this tested so far?

I only tested manually, by building the jar and running the VSCode plugin. 
I haven't looked into the existing tests yet, but I can take a look to see if we can add a test for this. Would it be specifically for testing the open liberty beta runtime?

File: lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/models/feature/Feature.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
-* Copyright (c) 2020, 2022 IBM Corporation and others.
+* Copyright (c) 2020-2023 IBM Corporation and others.
```
Comment (cherylking) [inline lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/models/feature/Feature.java:2]: Not sure why you used a different format here? It should be `2020, 2022`

**Review by cherylking — CHANGES_REQUESTED** [review]
Fix copyright format for each file. Also, the test would have to be with the beta runtime to force this code path. Unless you invent some unit test and check in a mock feature list xml file and make sure it loads correctly using the jaxb code?

**Comment by evie-lau** [comment]
Working on a unit test for using the beta runtime. Looking like I'll have to check in a `ws-featurelist.jar`.

**Review by scottkurz — CHANGES_REQUESTED** [review]
Looks good.  I tested it in Eclipse and it works.  I sent a PR to update the NOTICES coordinates https://github.com/evie-lau/liberty-language-server/pull/1

As far as testing.. I was just curious.   I'm not suggesting it's a MUST to merge this.

**Review by cherylking — APPROVED** [review]

**Review by scottkurz — APPROVED** [review]

**Pull request merged**